test/controllers/works_controller_test.rb

```diff
@@ -189,19 +220,64 @@
 
   describe "upvote" do
     it "redirects to the work page if no user is logged in" do
-      skip
+      work_id = works(:poodr)
+
+      expect {
+        post upvote_path(work_id)
+      }.wont_change "Vote.count"
+
+      must_respond_with :redirect
+      must_redirect_to root_path
     end
 
     it "redirects to the work page after the user has logged out" do
-      skip
+      perform_login
+
+      expect {
+        delete logout_path
+      }.wont_change "User.count"
+
+      must_respond_with :redirect
+      must_redirect_to root_path
     end
 
     it "succeeds for a logged-in user and a fresh user-vote pair" do
-      skip
+      perform_login
+      work_id = works(:poodr)
+
+      expect {
+        post upvote_path(work_id)
+      }.must_change "Vote.count", 1
+
+      must_respond_with :redirect
+      must_redirect_to work_path(work_id)
     end
 
-    it "redirects to the work page if the user has already voted for that work" do
-      skip
+    it "redirects to the individual work page if the user has already voted for that work" do
+      perform_login
+      work_id = works(:poodr)
+
+      expect {
+        post upvote_path(work_id)
+      }.must_change "Vote.count", 1
+
+      expect {
+        post upvote_path(work_id)
+      }.wont_change "Vote.count"
+
+      must_respond_with :redirect
+      must_redirect_to work_path(work_id)
+    end
+
+    it "redirects to the work page if a logged in user tries to upvote a work that does not exist" do
+      perform_login
+      work_id = 3490
```

Comment:
Just for note, it's likely that this ID doesn't exist.  It's not guaranteed.  Instead I suggest using -1, or deleting an existing work and then trying to upvote it.